### PR TITLE
Fixed multiple identical dependency IDs in qsub

### PR DESCRIPTION
**Pull request by vinjana**
Additionally removed array-job support and renamed BEJobDependencyID to BEJobID.

### CODE REVIEW

File: src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/pbs/PBSCommand.groovy [hunk 1]
```diff
@@ -231,71 +232,25 @@ class PBSCommand extends Command {
...
-                    depStrBld = new StringBuilder()
-                    depStrBld << tmp
-                }
-
-                String depStr = depStrBld.toString()
-                if (depStr.length() > 1) {
-                    qsubCall << getDependsSuperParameter() << depStr
-                }
+                qsubCall <<
+                        getDependsSuperParameter() <<
```
Comment (dankwart-de) [inline src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/pbs/PBSCommand.groovy]: I cannot find these methods??

File: src/main/groovy/de/dkfz/roddy/execution/jobs/BEJob.groovy [hunk 2]
```diff
@@ -121,7 +121,36 @@ class BEJob<J extends BEJob, JR extends BEJobResult> {
...
+
+    private static List<BEJobID> jobs2jobIDs(List<BEJob> jobs) {
+        if (null == jobs) {
+            return new LinkedList<BEJobID>()
+        } else {
+            return jobs.collect { it.runResult.jobID }
+        }
+    }
+
+    private List<BEJobID> reconcileParentJobInformation(List<BEJobID> parentJobIDs, List<BEJob> parentJobs) {
```
Comment (vinjana) [inline src/main/groovy/de/dkfz/roddy/execution/jobs/BEJob.groovy]: None of the three terms fits the meaning, though.

File: src/main/groovy/de/dkfz/roddy/execution/jobs/BEJob.groovy [hunk 3]
```diff
@@ -121,7 +121,36 @@ class BEJob<J extends BEJob, JR extends BEJobResult> {
...
+            if (a != b) {
+                throw new RuntimeException("parentJobBEJob needs to be called with one of parentJobs, parentJobIDs, or parentJobsIDs and *corresponding* parentJobs.")
+            }
+            pJids = findValidJobIDs(parentJobIDs)
+        } else if (null == parentJobIDs && null == parentJobs) {
+            pJids = new LinkedList<BEJobID>()
+        } else if (null != parentJobs) {
+            pJids = jobs2jobIDs(findJobsWithValidJobId(parentJobs))
+        } else {
+            pJids == findValidJobIDs(parentJobIDs)
```
Comment (vinjana) [inline src/main/groovy/de/dkfz/roddy/execution/jobs/BEJob.groovy]: Oops! 8-|

File: src/main/groovy/de/dkfz/roddy/execution/jobs/BEJob.groovy [hunk 4]
```diff
@@ -121,7 +121,36 @@ class BEJob<J extends BEJob, JR extends BEJobResult> {
...
+        }
+        return pJids;
+    }
+
+    BEJob(String jobName, File tool, String toolScript, String toolMD5, ResourceSet resourceSet, List<String> arrayIndices, Map<String, String> parameters, List<BEJob> parentJobs, List<BEJobID> parentJobIDs, BatchEuphoriaJobManager jobManager) {
```
Comment (vinjana) [inline src/main/groovy/de/dkfz/roddy/execution/jobs/BEJob.groovy]: O.k.

File: src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/pbs/PBSCommand.groovy (hunk 1 above)
Comment (vinjana) [inline src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/pbs/PBSCommand.groovy]: Good point. Idea finds them and apparently they are still in RoddyCore!

Comment (vinjana) [inline src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/pbs/PBSCommand.groovy]: l. 135, same file.

File: src/main/groovy/de/dkfz/roddy/execution/jobs/BEJob.groovy [hunk 5]
```diff
@@ -132,7 +132,7 @@ class BEJob<J extends BEJob, JR extends BEJobResult> {
         this.parameters = parameters
         this.parentJobs = parentJobs
         this.arrayIndices = arrayIndices ?: new LinkedList<String>()
-        this.listOfCustomDependencyIDs.addAll((dependencyIDs ?: parentJobs ? parentJobs.collect { BEJob job -> job.runResult?.jobID ?: null }.findAll { BEJobDependencyID it -> it } : []) as List<BEJobDependencyID>)
+        this.listOfCustomDependencyIDs.addAll(parentJobIDs)
```
Comment (fkaercher) [inline src/main/groovy/de/dkfz/roddy/execution/jobs/BEJob.groovy]: parentJobIDs can be null. Has to be checked.

**Pull request merged**